nestkernel/eprop_archiving_node.h

```diff
@@ -117,21 +199,99 @@ class EpropArchivingNodeRecurrent : public EpropArchivingNode< HistEntryEpropRec
   //! Copy constructor.
   EpropArchivingNodeRecurrent( const EpropArchivingNodeRecurrent& );
 
-  //! Create an entry in the eprop history for the given time step and surrogate gradient.
+  //! Select the surrogate gradient function.
+  double ( EpropArchivingNodeRecurrent::*select_surrogate_gradient(
+    std::string surrogate_gradient_function ) )( double, double, double, double, double, double )
```

Comment:
This code is rather messy. I have created a PR towards your branch to clean it up.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/3207/commits/64ff25ab14be1e371320bbbf9dd7cee3e4549230